templates/zerver/app/home.html

```diff
@@ -18,6 +18,16 @@
             {% endtrans %}
         </p>
     </div>
+    <div class="all-messages-search-caution" hidden>
+        <p>
+            <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
+            {% trans %}
+            This search may exclude some messages you have access to,
+            such as messages sent to public streams before you joined the stream.
+            <a href="/help" target="_blank">[Learn more]</a>
```

Comment:
@rishig I think the main thing blocking this is the "Learn more" link; probably that should just be to an anchor on the main search article?

Reply:
ah, I can just do it. 
and yeah, anchor on main search article is what I was thinking as well.

Reply:
Added in #13048 
The link should be to `/help/search-for-messages#messages-sent-before-you-joined`

Reply:
Made the change.